### ISSUE TITLE
Only one to three LEDs lights up and hangs

### ISSUE THREAD

**Issue by ArtursGailis1995** [jasoncoon/esp8266-fastled-webserver#69]
TLDR - I am having an issue getting this to work. Everything compiles and uploads fine, NodeMCU connects to my router and WEB interface loads all good, but WS2812B strip hangs (first LED comes on) and effects does not change.

I think I have tried everything and _I KNOW THAT MY HARDWARE WORKS_ because when I upload my own code to any of my NodeMCUs, it works well and LEDs change colors. I am using proper power supply and I have also tried using internal pins for power. I have common ground with the external power supply.

LED strip: https://www.ebay.com/itm/WS2812B-5050-RGB-LED-Strip-30-60-144-LEDs-M-ws2812-IC-Individual-Addressable-5V/262452005540?ssPageName=STRK%3AMEBIDX%3AIT&var=561255461889&_trksid=p2057872.m2749.l2649

NodeMCU #1: https://www.ebay.com/itm/ESP8266-NODEMCU-Preflashed-Development-Board-OLED-18650-Integrated-Circuits-New/222811187574?ssPageName=STRK%3AMEBIDX%3AIT&_trksid=p2057872.m2749.l2649

NodeMCU #2: https://www.ebay.com/itm/WIFI-Internet-Development-Board-based-on-ESP8266-Module-for-NodeMcu-Lua/400976251796?hash=item5d5c0c0b94:g:iPEAAOSwxp9W2~uJ

PSU: dont have link for that but its made by Hewlett-Packard and provides very stable 5V output and at high as 6 amps. Wires are as thick as laptop charger wires.

**What I have done so far:**
1) Installed latest Arduino IDE
2) Installed latest ESP8266 board in IDE (yes, I did try older version too)
3) Installed latest FastLED library
4) Installed latest IRremoteESP8266 (I dont even use IR yet)
5) Installed latest Arduino WebSockets
6) Installed latest Arduino ESP8266FS tool (yes, I did try older version too)
7) Choosed correct settings in IDE
8) Uploaded sketch data using ESP8266 Sketch Data Upload
9) Entered my routers SSID and password in WiFi file (WiFi works and WEBUI loads fine)
10) Changed my strip model to WS2812B, LED count to 300, tried different pins (tried different settings too)
11) Uploaded the sketch and tried to get it working
12) Changed config to start LED strip in Red color (fill_solid(leds, NUM_LEDS, CRGB::Red);) and it sometimes works, but the first LED is in wrong color.
13) Tried to use differnet ESP8266 model and repeated everything
14) After few more failures I repeated everything on different computer
15) Erased SPIFFS and EEPROM and tried again multiple times with different settings (on both PCs, multiple boards)
16) Tried connecting it to different router with different SSID and password
17) Checked my wiring 20 times and confirmed that it works with other sketches using FastLED library
18) Replaced all my wires and PSU just in case, tried shorter and longer, thicker and less thick wires
19) Tried to add configuration to disable interrupts (found this on Google)
20) Tried to add configuration to allow interrupt retries (fond this on Google too)
21) Looked at serial output dozen times
22) Started commenting out parts of the code related to EEPROM and setting loading (here I managed to somehow get it 50/50 working but the effects from WEBUI were really laggy and not consistent on all the strip)

What to do next? I really need to get this running. I dont need IR remote as of now, but I really need to be able to control it from my PC/Phone.

**Comment by maik-dmxc** [comment]
Hi,
i'm having the same problems, all uploadings are fine, but nothing on DataPin.
i'm getting this error while compiling
`In file included from /home/maik/Desktop/esp8266-fastled-webserver/esp8266-fastled-webserver.ino:19:0:
/home/maik/sketchbook/libraries/FastLED/FastLED.h:17:21: note: #pragma message: FastLED version 3.001.003
 #    pragma message "FastLED version 3.001.003"
                     ^
In file included from /home/maik/sketchbook/libraries/FastLED/FastLED.h:65:0,
                 from /home/maik/Desktop/esp8266-fastled-webserver/esp8266-fastled-webserver.ino:19:
/home/maik/sketchbook/libraries/FastLED/fastspi.h:110:23: note: #pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output
 #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
                       ^`
webserver runs perfect, but strip do nothing!

using Nodemcu 1.0

regards maik

**Comment by maik-dmxc** [comment]
ok,
I downloaded the master with Latest commit 04103b8 on 18 Mar
it work with D1mini, but have big problems with 1200 leds.
the first 255 are fine , the next leds are flickering heavy.

if you will, i can make a video

regards maik

**Comment by jasoncoon** [comment]
@maik-dmxc tried another, simpler FastLED or NeoPixel example?  Using a level shifter?

**Comment by maik-dmxc** [comment]
tried all examples of FastLed library, all the same issue. first 250-300 are fine , the next 1000 pixels are awefull. flickering and jumping 1 pixel forward and back.
so problem is FastLed library.

also tried WS2812FX library, all fine 1200 leds are working smothly.

regards maik

**Comment by maik-dmxc** [comment]
found solution
add #define FASTLED_ALLOW_INTERRUPTS 0 to the line immediately before you have #include <FastLED.h> - FastLED is different from the other libraries in that it allows interrupts to fire while processing WS2812 leds - however, if an interrupt takes too long to run, then it has to stop writing 

now everything works smothly without flickering!

regards maik

**Comment by jasoncoon** [comment]
Anyone still having issues with jitter / flickering should try these suggestions as well: https://github.com/jasoncoon/esp8266-fastled-webserver/issues/85#issuecomment-427595122

**Comment by leded** [comment]
I had this problem too, or at least a very similar one.  I solved it by connecting the ground of the nodemcu to the same ground as the led light strip.  Powering the nodemcu off of the same psu isn't enough, it needs to be the same terminals as the led strip.
